src/components/sbb-link/sbb-link.tsx

```diff
@@ -63,88 +76,134 @@ export class SbbLink {
-   * Choose the link style variant.
+   * Applies link inline styles (underline, inherit coloring/font-size etc).
    */
-  @Prop() public variant: InterfaceLinkAttributes['variant'] = 'block';
+  @Prop() public inline = false;
 
-  private get _inlineVariant(): boolean {
-    return this.variant === 'inline' || this.variant === 'inline-negative';
-  }
+  /**
+   * Disabled attribute if link is used as button (optional)
+   */
+  @Prop() public disabled?: boolean;
+
+  /**
+   * Name attribute if link is used as button (optional)
+   */
+  @Prop() public name?: string;
+
+  /**
+   * Form attribute if link is used as button (optional)
+   */
+  @Prop() public form?: string;
+
+  /**
+   * Type attribute if link is used as button (optional)
+   */
+  @Prop() public type?: InterfaceLinkAttributes['buttonType'];
 
+  /**
+   * Render element
+   */
   public render(): JSX.Element {
-    const textSizeClass = this._inlineVariant ? '' : ` sbb-link--text-${this.textSize}`;
+    let TAG_NAME;
+    if (this.isStatic) {
+      TAG_NAME = 'span';
+    } else if (this.href) {
+      TAG_NAME = 'a';
+    } else {
+      TAG_NAME = 'button';
+    }
+
+    return (
+      <TAG_NAME {...this._getAttributeList}>
+        {this.icon && !this.inline ? (
+          <span class="sbb-link__icon">
+            <slot name="icon" />
+          </span>
+        ) : (
+          ''
+        )}
+        <slot />
+      </TAG_NAME>
+    );
+  }
+
+  /**
+   * Get the attributelist base on the config and the resulting element
+   * @private
+   * @return <object>
+   */
+  private get _getAttributeList(): object {
+    let ariaLabelText = this.accessibilityLabel;
     const currentLanguage = getDocumentLang();
     const currentWritingMode = getDocumentWritingMode();
+    const attributeList: Record<string, string> = {};
+
+    Object.assign(attributeList, {
+      dir: currentWritingMode || undefined,
+      class: this._getClassString || undefined,
+      'aria-label': ariaLabelText || undefined,
+      name: this.name || undefined,
+      id: this.idValue || undefined,
+    });
 
     let openInNewWindow = false;
 
-    if (!window.location.href.includes(this.hrefValue)) {
+    if (!window.location.href.includes(this.href)) {
       openInNewWindow = true;
     }
 
-    /**
-     * Add additional CSS classes
-     * ----------------------------------------------------------------
-     */
-    let iconPositionClass = '';
+    if (openInNewWindow && !this.isStatic && this.href) {
+      ariaLabelText += `. ${i18nTargetOpensInNewWindow[currentLanguage]}`;
 
-    if (this.icon) {
-      iconPositionClass = ` sbb-link--icon-placement-${this.iconPlacement}`;
+      Object.assign(attributeList, {
+        rel: 'external noopener nofollow',
+        target: '_blank',
+        'aria-label': ariaLabelText || undefined,
+      });
     }
 
-    let iconFlipClass = '';
-
-    if (this.icon && this.iconFlip) {
-      iconFlipClass = ' sbb-link--icon-flip';
+    if (!this.isStatic) {
```

Comment:
I'd prefer to avoid if nesting. Could we do an early return if it is static?

Reply:
There is no if-nesting and this condition is related to the icon position. Not if the element appears as a span.

Reply:
With if-nesting I mean there is an if inside an if.
This could be refactored to:
```ts
if (this.isStatic) {
  return attributeList;
}
```

Reply:
You were at the wrong line. ;)  But found the pos that you mentioned.